test/media/streaming_engine_unit.js

```diff
@@ -3041,6 +3041,33 @@ describe('StreamingEngine', () => {
       expect(secondRequest.abort).toHaveBeenCalled();
     });
 
+    it('aborts segment without size', async () => {
+      const videoStream = manifest.periods[0].variants[0].video;
+      await videoStream.createSegmentIndex();
+      const segmentIndex = videoStream.segmentIndex;
```

Comment:
You should set `videoStream.bandwidth` to a positive integer. Something small like 10 is fine. Our mock streams have a bandwidth of 0 by default, and the code in the fix does multiply by bandwidth.